stable/spark-history-server/templates/deployment.yaml

```diff
@@ -60,31 +69,48 @@ spec:
           httpGet:
             path: /
             port: historyport
-        volumeMounts:
```

Comment:
If you leave the `volumeMounts` here, then you don't have to write it multiple times below.

Reply:
This was done intentionally. There are two cases within when s3 is enabled: (1) IAM authentication is used (2) IAM authentication is NOT used and fall back to using access key id and secret key.

I do not need `volumeMounts` or `volumes` for the former case. I only need the secret volume to be mounted in the later case. If I left the `volume` and `volumeMounts` at the topmost level, the former case will produce the spec file with empty `volume` and `volumeMounts` as below, which results in an error while deploying:
```
volumeMounts:
<empty>
volumes:
<empty>
```

Reply:
Got it! Thanks for the clarification.